### PR TITLE
1679 Add icon support to va-accordion-item

### PR DESCRIPTION
**Pull request by mdewey**
## Chromatic

https://1679-accordion-header-update--60f9b557105290003b387cd5.chromatic.com

## Description
Closes [Ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1679)
Address UX updates needed for [secure messaging](https://www.sketch.com/s/32bdb345-fb1f-409b-985f-ff627cb60032)
## Testing done
- [ ] Chrome
- [x] Storybook
- [ ] Various Screen sizes
- [x] updated unit testing

## Screenshots
image
image

## Acceptance criteria
- [ ] Icons and bolding are in the header

## Definition of done
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

## Important notes

- I am looking for feedback on the approach, [context](https://github.com/department-of-veterans-affairs/component-library/pull/674#discussion_r1164625345), before fixing the failing tests, polishing the code and polishing the docs. 
- I am still working on the docs page and cleaning up the samples. but wanted to get the review process started while I do that.

### CODE REVIEW

File: packages/web-components/src/components/va-accordion/va-accordion-item.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-accordion/va-accordion-item.tsx:116]: Just because it's a named slot doesn't mean it's limited to just icons unless we programmatically only allow icon nodes to be passed into it but if @humancompanion-usds is fine with this approach, I don't have any issues with it. I think it makes sense to have it a slot since we're unsure of which icon library would be used here in the future.

Minor style question: Is it necessary to have `{ }` around the slot?

File: packages/web-components/src/components/va-accordion/va-accordion.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-accordion/va-accordion.tsx:74]: Is this essentially just an IE cleanup/refactor?

File: packages/web-components/src/components/va-accordion/va-accordion-item.tsx
Comment (humancompanion-usds) [reply packages/web-components/src/components/va-accordion/va-accordion-item.tsx:116]: I defer to you both. Seems fine to me.

File: packages/web-components/src/components/va-accordion/va-accordion.tsx
Comment (mdewey) [reply packages/web-components/src/components/va-accordion/va-accordion.tsx:74]: Basically. On that topic, what browsers do we need to support?

File: packages/web-components/src/components/va-accordion/va-accordion-item.tsx
Comment (mdewey) [reply packages/web-components/src/components/va-accordion/va-accordion-item.tsx:116]: Minor style answer: Those can be removed, It was a ternary thing there before, but that turned out not to be needed.

File: packages/web-components/src/components/va-accordion/va-accordion.tsx
Comment (jamigibbs) [reply packages/web-components/src/components/va-accordion/va-accordion.tsx:74]: > that browsers do we need to support?

I forgot to answer your actual question! Here is our current [browser support policy](https://vfs.atlassian.net/wiki/spaces/pilot/pages/1847951366/Browser+Support+Policy):

Screenshot 2023-04-20 at 10 10 53 AM

File: packages/web-components/package.json
Comment (jamigibbs) [inline packages/web-components/package.json:3]: If we're doing a minor bump, let's kick this to 4.34.0

File: packages/web-components/src/components/va-accordion/va-accordion.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-accordion/va-accordion.tsx:84]: Any particular reason that the [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) `?` was removed in various places?

File: packages/web-components/src/components/va-accordion/va-accordion.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-accordion/va-accordion.tsx:75]: Remove console log.

File: README.md
Comment (jamigibbs) [inline README.md:119]: Thank you (again)!

**Review by jamigibbs — COMMENTED** [review]
A lot of nice additions here! Thanks again for your help. 

I left just a few comments and also can you double check the responsive view for long headline text? I suspect we will not want the text to wrap underneath the icon like this but I'll let @humancompanion-usds confirm:

Screenshot 2023-04-21 at 4 24 58 PM

It might be as simple as adding `display: flex` to the button:

Screenshot 2023-04-21 at 4 31 12 PM

Screenshot 2023-04-21 at 4 33 12 PM

**Comment by mdewey** [comment]
@jamigibbs I have addressed all your comments (great catches! ). Let me know what else I need to do here.  Once this is merged I will update vets-website

**Review by jamigibbs — APPROVED** [review]
This looks great, thanks Mark! We will need to create a new component library release in order to make this available in vets-website which isn't a problem but before merging, could you:

1. Update the `core` package version to 15.2.0
2. Update the PR title to something more descriptive to the actual update (this will help us with the release notes). ie. Add icon support to va-accordion-item

**Pull request merged**
